src/PropertyInfo/TypeInfoParsers/NativeTypeInfoParser.php

```diff
@@ -19,11 +17,8 @@
  *
  * @author Mihai Stancu <stancu.t.mihai@gmail.com>
  */
-abstract class NativeTypeInfoParser implements TypeInfoParserInterface
+trait NativeTypeInfoParser
```

Comment:
Purely technical it is a BC break but the class has only been present in the repo for a month (I added the abstract). To avoid this possible BC break I could rename the traits with a `Trait` suffix and create an `abstract class NativeTypeInfoParser` which uses the trait.

The second branch has many more BC breaks. Perhaps we should about applying `@api` doc comments on publicly extendable classes (as the Symfony standard does) and `@internal` on others.